hare/hare.go

```diff
@@ -241,8 +255,9 @@ func (h *Hare) onTick(ctx context.Context, id types.LayerID) {
 	// retrieve set from orphan blocks
 	blocks, err := h.msh.LayerBlockIds(h.lastLayer)
 	if err != nil {
-		logger.With().Error("no blocks for consensus", log.Err(err))
-		return
+		// we have to try achieve consensus on empty set or fail, otherwise it will be the delayed consensus process forever
```

Comment:
I don't understand this comment. Why would it "otherwise delay the consensus process forever"? If hare never runs for layer N, it will still try to run for layer N+1

Reply:
It was before I added defer finishing hare with "notCompleted" state.